torch/_inductor/fx_passes/joint_graph.py

```diff
@@ -29,9 +29,11 @@ def lazy_init():
     from .fuse_attention import _sfdp_init
     from .misc_patterns import _misc_patterns_init
     from .pad_mm import _pad_mm_init
+    from .woq_mm import _woq_mm_init
 
     _pad_mm_init()
     _sfdp_init()
+    _woq_mm_init()
```

Comment:
no, right now the int8 woq is a mixed dtype linear and int8 is dequantized to float for computation. So we do not need to do weight prepacking right now.

Reply:
But eventually we would need prepacking, right? I guess we would assume weights are frozen here, right?

Reply:
Yes. Prepacking will be needed for amx-int8 and avx512-vnni, but not for now.

Reply:
Thanks. As we synced offline, please move the pass to freezing passes.

Reply:
As synced offline, since we do not need weight prepacking right now and moving to freezing passe requires much effort, we would temporarily keep the current passe to catch up with PT2.3. cc @jgong5

Reply:
Please comment on why cuda doesn't need this pass.

Reply:
Without this pass, cuda actually has the same behavior as cpu:
1) calls `at::matmul` in `linear`;
2) decomposes from `matmul` to `bmm`;
3) decomposes from `bmm` to `2unsqueeze+mul+sum`.

```
# File: /home/eikan/local_disk/liaoxuan/gpt-fast/quantize.py:355 in forward, code: return F.linear(input, self.weight.to(dtype=input.dtype)) * self.scales
        convert_element_type_2: "bf16[12288, 4096]" = torch.ops.prims.convert_element_type.default(arg68_1, torch.bfloat16);  arg68_1 = None
        permute: "bf16[4096, 12288]" = torch.ops.aten.permute.default(convert_element_type_2, [1, 0]);  convert_element_type_2 = None
        view: "bf16[1, 4096]" = torch.ops.aten.view.default(mul_2, [1, 4096]);  mul_2 = None
        convert_element_type_3: "f32[1, 4096]" = torch.ops.prims.convert_element_type.default(view, torch.float32);  view = None
        convert_element_type_4: "f32[4096, 12288]" = torch.ops.prims.convert_element_type.default(permute, torch.float32);  permute = None
        unsqueeze_2: "f32[1, 4096, 1]" = torch.ops.aten.unsqueeze.default(convert_element_type_3, 2);  convert_element_type_3 = None
        unsqueeze_3: "f32[1, 4096, 12288]" = torch.ops.aten.unsqueeze.default(convert_element_type_4, 0);  convert_element_type_4 = None
        mul_3: "f32[1, 4096, 12288]" = torch.ops.aten.mul.Tensor(unsqueeze_2, unsqueeze_3);  unsqueeze_2 = unsqueeze_3 = None
        sum_1: "f32[1, 12288]" = torch.ops.aten.sum.dim_IntList(mul_3, [1]);  mul_3 = None
        convert_element_type_5: "bf16[1, 12288]" = torch.ops.prims.convert_element_type.default(sum_1, torch.bfloat16);  sum_1 = None
        view_1: "bf16[1, 1, 12288]" = torch.ops.aten.view.default(convert_element_type_5, [1, 1, 12288]);  convert_element_type_5 = None
        mul_4: "bf16[1, 1, 12288]" = torch.ops.aten.mul.Tensor(view_1, arg69_1);  view_1 = arg69_1 = None
```
cc @jgong5 @mingfeima

Reply:
Can we move to freezing passes ? This should not be a huge refactor. I would we didnt take short term technical debt to hit release. They are still accepting cherry-picks anyway.